libc/getpw.c

```diff
@@ -0,0 +1,111 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libc-tests
+ *
+ * Testing getpwuid and getpwnam edge cases
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Mateusz Niewiadomski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+/* make compilable against glibc (all these tests were run on host against libc/linux kernel) */
+#define _GNU_SOURCE
+
+#include <pwd.h>
+#include <unistd.h>
+
+#include <unity_fixture.h>
+
+
+static unsigned int ispasswdfile;
+static const char root[] = "root";
+static struct passwd *pw;
+
+
+TEST_GROUP(getpwd);
+
+
+TEST_SETUP(getpwd)
+{
+	ispasswdfile = (access("/etc/passwd", F_OK) == 0) ? 1 : 0;
```

Comment:
You know that this gets called before every test? Maybe You should do it once in `TEST_GROUP_RUNNER` - and depending on the result either run every test with result "SKIPPED" or don't run any tests at all.

Reply:
Thanks for that idea. Moved the `ispasswdfile` flag initialization to `TEST_GROUP_RUNNER`